coremltools/models/utils.py

```diff
@@ -2251,4 +2252,199 @@ def _make_second_chunk_prog(prog: _mil.Program, op_idx: int) -> _mil.Program:
 
     return prog
 
+def change_input_output_tensor_type(
+    ml_model: "_ct.models.model.MLModel",
+    from_type: "_ct.proto.FeatureTypes_pb2.ArrayFeatureType",
+    to_type: "_ct.proto.FeatureTypes_pb2.ArrayFeatureType",
+    function_names: _Optional[_List[str]] = None,
+    input_names: _Optional[_List[str]] = None,
+    output_names: _Optional[_List[str]] = None,
+) -> "_ct.models.model.MLModel":
+    """
+    Change output type of the Core ML model tensor outputs. Supported types are FLOAT16, FLOAT32.
+
+    Parameters
+    ----------
+    ml_model: MLModel
+        A Core ML model that needs to change its output type.
+        Note:
+        - the original model is not modified, the model with updated output types is returned as a new instance.
+        - only an mlProgram is supported (not pipelines, not neural networks).
+
+    from_type:
+        The output type that should be changed from.
+
+    to_type:
+        The output type that will be used instead of all the outputs of `from_type` type.
+
+    function_names:
+        Optional list of function names where the output needs to be changed. If not specified, only the "main"
+        function will be updated.
 
+    Examples
+    --------
+    .. sourcecode:: python
+
+        from coremltools.models.model import MLModel
+        from coremltools.utils import change_array_output_type
+        from coremltools.proto.FeatureTypes_pb2 import ArrayFeatureType
+
+        model = MLModel("my_model.mlpackage")
+        updated_model = change_output_tensor_type(
+            ml_model=model,
+            from_type=ArrayFeatureType.FLOAT32,
+            to_type=ArrayFeatureType.FLOAT16,
+        )
+        updated_model.save("my_updated_model.mlpackage")
+    """
+    # We do the lazy import to prevent circular import
+    from coremltools.converters.mil.converter import mil_convert as _mil_convert
+    from coremltools.converters.mil import Var, Operation
+    from coremltools.converters.mil.mil.block import Function
+    from coremltools.converters.mil.mil.types import fp16, fp32, tensor, double
```

Comment:
We may not want to import those types directly. It's recommended to only import `types` and use `types.fp16` etc to access those types.

Reply:
Updated, re-used an other existing import for these types references.